src/main/java/io/openapitools/swagger/GenerateMojo.java

```diff
@@ -152,14 +156,34 @@ private Application resolveApplication(JaxRSScanner reflectiveScanner) {
         Class<? extends Application> appClazz = (Class<? extends Application>)clazz;
         return ClassUtils.createInstance(appClazz);
     }
-    
+
     private URLClassLoader createClassLoader() {
         try {
-            File compiled = new File(project.getBuild().getOutputDirectory());
-            return new URLClassLoader(new URL[] {compiled.toURI().toURL()}, Thread.currentThread().getContextClassLoader());
+            Collection<String> dependencies = getDependentClasspathElements();
+            URL[] urls = new URL[dependencies.size()];
+            int index = 0;
+            for (String dependency : dependencies) {
+                urls[index++] = Paths.get(dependency).toUri().toURL();
+            }
+            return new URLClassLoader(urls, Thread.currentThread().getContextClassLoader());
         } catch (MalformedURLException e) {
             throw new RuntimeException("Unable to create class loader with compiled classes", e);
+        } catch (DependencyResolutionRequiredException e) {
+            throw new RuntimeException("Dependency resolution (runtime + compile) is required");
+        }
+    }
+
+    private Collection<String> getDependentClasspathElements() throws DependencyResolutionRequiredException {
+        Set<String> dependencies = new LinkedHashSet<>();
+        Collection<String> compileClasspathElements = project.getCompileClasspathElements();
+        if (compileClasspathElements != null) {
```

Comment:
Wondering if this can ever be `null` or if the `DependencyResolutionRequiredException` would then be thrown. But the documentation does not seem to say anything about this so better keep the null check.

Reply:
According to the implementation is should not, but there is no documentation that explicitly states this, and I did not want to base this just on the implementation details of `MavenProject`. 

Additionally, the `MavenProjectStub` returns `null` by default from this method, so this check is also a matter of convenience (otherwise some altering of the test setup would be needed).